hhvm/fbexpect
# Add assert and remove phpunit

I apologize in advance -- this diff is huge but all of these pieces need to be in place simultaneously in order to remove PHPUnit as a dependency entirely.

- PHPUnit is gone! Using HackTest to run tests
- Most of the logic for asserts is from PHPUnit, so nothing too special
- Added hh_autoload
- Added custom exception classes
- Removed one test that used toEqualWithDelta with arrays (restrict to num)

TODO (in another PR):
- Use hacktest in travis (this travis build will fail)
- Update README

Once this lands I can:
- Remove PHPUnit as a dependency from HackTest
- Check for my custom exceptions (rather than the PHPUnit versions) in external version of HackTest (www has its own version of expect)
- Remove PHPUnit from HSL 
- Submit diff to refactor HSL to use HackTest (basically ready, waiting for this)
- After above, update all three repos' (HackTest, HSL, FBExpect) composer.lock so that PHPUnit stuff doesn't show up anymore
